Packages/ClientRuntime/Sources/Networking/Http/Middlewares/SerializableBodyMiddleware.swift

```diff
@@ -0,0 +1,44 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+public struct SerializableBodyMiddleware<OperationStackInput: Encodable & Reflection,
+                                         OperationStackOutput: HttpResponseBinding,
+                                         OperationStackError: HttpResponseBinding>: Middleware {
+    public let id: Swift.String = "\(String(describing: OperationStackInput.self))BodyMiddleware"
+    
+    let alwaysSendBody: Bool
+    
+    public init(alwaysSendBody: Bool = false) {
+        self.alwaysSendBody = alwaysSendBody
+    }
+    
+    public func handle<H>(context: Context,
+                          input: SerializeStepInput<OperationStackInput>,
+                          next: H) -> Swift.Result<OperationOutput<OperationStackOutput>, MError>
+    where H: Handler,
+          Self.MInput == H.Input,
+          Self.MOutput == H.Output,
+          Self.Context == H.Context,
+          Self.MError == H.MiddlewareError {
+              do {
+                  if try !input.operationInput.allPropertiesAreNull() || alwaysSendBody {
```

Comment:
We should switch the order of `alwaysSendBody` with `try !input.operationInput.allPropertiesAreNull()`

Reply:
corected